src/app/pages/applications/forms/chart-form.component.ts

```diff
@@ -0,0 +1,343 @@
+          });
+        } else {
+          fieldSet.config = fieldSet.config.concat(this.getFieldConfigs(question));
+        }
+      }
+    });
+
+    this.fieldSets = this.fieldSets.filter(fieldSet => fieldSet.config.length > 0);
+    
+  }
+
+  parseConfigData(configData, parentKey, result) {
+    Object.keys(configData).forEach(key => {
+      const value = configData[key];
+      let fullKey = key;
+      if (parentKey) {
+        fullKey = `${parentKey}_${key}`;
+      }
+      if (!Array.isArray(value) && typeof value === 'object') {
+        this.parseConfigData(value, fullKey, result);
+      } else {
+        result[fullKey] = value;
+      }
+    });
+  }
+
+  resourceTransformIncomingRestData(data) {
+    
+    this.name = data.name;
+    const configData = {};
+    this.parseConfigData(data.config, null, configData);
+    configData['release_name'] = data.name;
+    
+    return configData;
+  }
+
+  afterInit(entityEdit: any) {
+    if (this.rowName) {
+      entityEdit.setDisabled('release_name', true, false);
+    }
+
+    let repositoryConfig = _.find(this.fieldConfig, {'name': 'image_repository'});
+    if (repositoryConfig) {
+      repositoryConfig.readonly = true;
+    }
+    
+  }
+
+  setObjectValues(data, result) {
+    Object.keys(data).forEach(key => {
+      const key_list = key.split('_');
+      const value = data[key];
+      if (key == "release_name" || value == "" || value == undefined) {
+        return;
+      }
+      
+      if (key_list.length > 1) {
+        let parent = result;
+        for(let i=0; i<key_list.length; i++) {
+          const temp_key = key_list[i];
+          if (i == key_list.length - 1) {
+            if (Array.isArray(value)) {
+              const arrayValues = this.setArrayValues(value);
+              if (arrayValues.length > 0) {
+                parent[temp_key] = arrayValues;
+              }
+            } else {
+              parent[temp_key] = value;
+            }            
+          } else {
+            if (!parent[temp_key]) {
+              parent[temp_key] = {};
+            }
+            parent = parent[temp_key];
+          }
+        }        
+      } else {
+        if (Array.isArray(value)) {
+          const arrayValues = this.setArrayValues(value);
+          if (arrayValues.length > 0) {
+            result[key] = arrayValues;
+          }
+          
+        } else {
+          result[key] = value;
+        }
+      }
+      
+    });
+
+    return result;
+  }
+
+  setArrayValues(data) {
+    const arrayVaules = [];
+    data.forEach(item => {
+      const oneItem = {};
+      this.setObjectValues(item, oneItem);
+      if (Object.keys(oneItem).length > 0) {
+        arrayVaules.push(oneItem);
+      }      
+    });
+
+    return arrayVaules;
+  }
+
+  customSubmit(data) {
+    let apiCall = this.addCall;
+    let values = {};
+    this.setObjectValues(data, values);
+
+    let payload = [];
+    payload.push({
+      catalog: 'OFFICIAL',
```

Comment:
I think this could already be based on the origin catalog...
There are already more hardcoded references to "official" than strictly needed....